pkg/inject/inject.go

```diff
@@ -336,71 +341,70 @@ func (conf *ResourceConfig) complete(template *v1.PodTemplateSpec) {
 
 // injectPodSpec adds linkerd sidecars to the provided PodSpec.
 func (conf *ResourceConfig) injectPodSpec(patch *Patch) {
+	if !conf.globalConfig.GetCniEnabled() {
+		conf.injectProxyInit(patch)
+	}
+
 	proxyUID := conf.proxyUID()
 	sidecar := v1.Container{
 		Name:                     k8s.ProxyContainerName,
 		Image:                    conf.taggedProxyImage(),
 		ImagePullPolicy:          conf.proxyImagePullPolicy(),
 		TerminationMessagePolicy: v1.TerminationMessageFallbackToLogsOnError,
-		SecurityContext: &v1.SecurityContext{
-			RunAsUser: &proxyUID,
-		},
+		SecurityContext:          &v1.SecurityContext{RunAsUser: &proxyUID},
 		Ports: []v1.ContainerPort{
-			{
-				Name:          k8s.ProxyPortName,
-				ContainerPort: conf.proxyInboundPort(),
-			},
-			{
-				Name:          k8s.ProxyMetricsPortName,
-				ContainerPort: conf.proxyMetricsPort(),
-			},
+			{Name: k8s.ProxyPortName, ContainerPort: conf.proxyInboundPort()},
+			{Name: k8s.ProxyAdminPortName, ContainerPort: conf.proxyAdminPort()},
 		},
 		Resources: conf.proxyResourceRequirements(),
 		Env: []v1.EnvVar{
 			{
-				Name:  envVarProxyLog,
+				Name:  envLog,
 				Value: conf.proxyLogLevel(),
 			},
 			{
-				Name:  envVarProxyControlURL,
-				Value: conf.proxyControlURL(),
+				Name:  envDestinationSvcAddr,
+				Value: conf.proxyDestinationAddr(),
 			},
 			{
-				Name:  envVarProxyControlListener,
-				Value: conf.proxyControlListener(),
+				Name:  envControlListenAddr,
+				Value: conf.proxyControlListenAddr(),
 			},
 			{
-				Name:  envVarProxyMetricsListener,
-				Value: conf.proxyMetricsListener(),
+				Name:  envAdminListenAddr,
+				Value: conf.proxyAdminListenAddr(),
 			},
 			{
-				Name:  envVarProxyOutboundListener,
-				Value: conf.proxyOutboundListener(),
+				Name:  envOutboundListenAddr,
+				Value: conf.proxyOutboundListenAddr(),
 			},
 			{
-				Name:  envVarProxyInboundListener,
-				Value: conf.proxyInboundListener(),
+				Name:  envInboundListenAddr,
+				Value: conf.proxyInboundListenAddr(),
 			},
 			{
-				Name:  envVarProxyDestinationProfileSuffixes,
+				Name:  envDestinationProfileSuffixes,
 				Value: conf.proxyDestinationProfileSuffixes(),
 			},
 			{
-				Name:      envVarProxyPodNamespace,
-				ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.namespace"}},
+				Name:  envInboundAcceptKeepAlive,
+				Value: fmt.Sprintf("%dms", defaultKeepaliveMs),
 			},
 			{
-				Name:  envVarProxyInboundAcceptKeepAlive,
+				Name:  envOutboundConnectKeepAlive,
 				Value: fmt.Sprintf("%dms", defaultKeepaliveMs),
 			},
 			{
-				Name:  envVarProxyOutboundConnectKeepAlive,
-				Value: fmt.Sprintf("%dms", defaultKeepaliveMs),
+				Name:      "K8S_NS",
```

Comment:
Nit: make this as a `const`, to be consistent with the others env vars?

Reply:
I chose not to make this one a const since it is effectively a "temp variable" that is only used  for templating. Is there a better way to connote this? perhaps call it `_k8s_ns` or something?

Reply:
Sure, I think prefixing this and the [rest](https://github.com/linkerd/linkerd2/blob/ver/identikit/pkg/inject/inject.go#L484-L554) with an underscore will communicate to future readers that this is intentional.